patches/net/minecraft/data/registries/RegistriesDatapackGenerator.java.patch

```diff
@@ -33,11 +41,29 @@
                              .flatMap(
                                  p_256552_ -> this.dumpRegistryCap(p_255785_, p_326736_, dynamicops, (RegistryDataLoader.RegistryData<?>)p_256552_).stream()
                              )
-@@ -54,6 +_,7 @@
+@@ -54,13 +_,14 @@
                      PackOutput.PathProvider packoutput$pathprovider = this.output.createRegistryElementsPathProvider(resourcekey);
                      return CompletableFuture.allOf(
                          p_349921_.listElements()
 +                            .filter(holder -> this.namespacePredicate.test(holder.key().location().getNamespace()))
                              .map(
                                  p_256105_ -> dumpValue(
                                          packoutput$pathprovider.json(p_256105_.key().location()),
+                                         p_256502_,
+                                         p_256000_,
+-                                        p_256449_.elementCodec(),
+-                                        p_256105_.value()
++                                        net.neoforged.neoforge.common.conditions.ConditionalOps.createConditionalCodecWithConditions(p_256449_.elementCodec()),
```

Comment:
You can create the codec outside of the lambda to avoid creating it for every element.